docs/blog/_posts/2019-05-24-15th-dotty-milestone-release.md

```diff
@@ -0,0 +1,216 @@
+
+## `given` clause comes last
+In the previous release, you could write something like this:
+
+```scala
+implied for String = "foo"
+def f(x: Int) given (y: String) (z: Int) = x + z
+f(1)(3)
+```
+
+Now, however, `given` clauses must come last. The above code will fail with:
+
+```
+-- Error: ../issues/Playground.scala:3:34 --------------------------------------
+3 |  def f(x: Int) given (y: String) (z: Int) = x + z
+  |                                  ^
+  |                       normal parameters cannot come after `given' clauses
+one error found
+```
+
+The following snippet is the correct way to express the program in question:
+
+```scala
+implied for String = "foo"
+def f(x: Int)(z: Int) given (y: String) = x + z
+f(1)(3)
+```
+
+We changed this to reduce confusion when calling functions with mixed explicit and implied parameters.
+
+## Type-safe Pattern Bindings
+```scala
+  val xs: List[Any] = List(1, 2, 3)
+  val (x: String) :: _ = xs   // error: pattern's type String is more specialized
+                              // than the right hand side expression's type Any
+```
+
+The above code will fail with a compile-time error in Scala 3.1 and in Scala 3 with the `-strict` flag. In contrast, in Scala 2, the above would have compiled fine but failed on runtime with an exception.
+
+Dotty compiler will allow such a pattern binding only if the pattern is *irrefutable* – that is, if the right-hand side conforms to the pattern's type. E.g. the following is OK:
+
+```scala
+  val pair = (1, true)
+  val (x, y) = pair
+```
+
+If we want to force the pattern binding if the pattern is not irrefutable, we can do so with an annotation:
+
+```scala
+  val xs: List[Any] = List("1", "2", "3")
+  val (x: String) :: _: @unchecked = xs
+```
+
+The same is implemented for pattern bindings in `for` expressions:
+
+```scala
+  val elems: List[Any] = List((1, 2), "hello", (3, 4))
+  for ((x, y) <- elems) yield (y, x) // error: pattern's type (Any, Any) is more specialized
+                                     // than the right hand side expression's type Any
+```
+
+For the migration purposes, the above change will only take effect in Scala 3.1. You can use it in Scala 3 with the `-strict` flag.
+
+For more information, see the [documentation](http://dotty.epfl.ch/docs/reference/changed-features/pattern-bindings.html).
+
+## Further improvements to Generalised Algebraic Data Types (GADTs) support
+In this release, we've further improved our support for GADTs. Most notably, we now support variant GADTs:
+
+```scala
+enum Expr[+T] {
+  case StrLit(s: String) extends Expr[String]
+  case Pair[A, B](a: Expr[A], b: Expr[B]) extends Expr[(A, B)]
+}
+
+def eval[T](e: Expr[T]): T = e match {
+  case Expr.StrLit(s) => s
+  case Expr.Pair(a, b) => (eval(a), eval(b))
+}
+```
+
+We've also plugged a few soundness problems caused by inferring too much when matching on abstract, union and intersection types.
```

Comment:
Maybe add the PR number (this right https://github.com/lampepfl/dotty/issues/2985)?

Reply:
@AleksanderBG is that issue correct here?

Reply:
It'd be better placed in the first line, which talks about variant GADTs. We can mention #5667 for an example of the soundness problems as well, though I didn't really create an issue for that.